### DISCUSSION TITLE
Add support for webgl 1.0 in godot 4

### DISCUSSION THREAD

**Discussion by Lortemanden** [godotengine/godot-proposals#11239] [Platforms]
Godot 4 only supports webgl 2.0 for webexports even though it is not usable for all major browsers. As stated in the [doc ](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_web.html) it does not work on macOs and iOS because of some bugs. Not being able to support iOS makes godot 4 unviable as a engine for developing games to the web, as it cuts off a huge userbase. And this is not just for commercial use godot 4 becomes unviable, since I have heard people not using it in game jams or their hobby games for this exact reason. 

While, for my understanding, the bugs with webgl 2.0 lies with Apple they seem uninterested in fixing it, since the problem has been known for years. So while it is frustrating if godot 4 wants to be viable for targeting the web it cant hope for Apple to come fix the problem. Who knows how many years will pass before webgl 2.0 becomes viable? WebGPU is meant to replace webgl 2.0 but only few browsers support it and who knows when it will not just be "supported" but actually work on all major browers?

My propasal is therefor to add support back for webgl 1.0 since it works for all major browsers.

**Comment by AThousandShips** [comment]
Would webgl 1.0 be able to support the needed features in Godot 4? And would it make Godot 4 relevant for this instead of just exporting for the relevant platforms directly or using Godot 3?

**Comment by Lortemanden** [comment]
I dont know if there is any technical limitations for Godot 4 to support webgl 1.0 but I would say it is relevant if Godot 4 wants to be a viable option for web exports. Yea you can use Godot 3 but that seems like a short sighted solution that just pushes the problem down the road.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/10044.

Regarding macOS/iOS-specific issues, we can find and implement workarounds while still using WebGL 2.0. For instance, many problems go away if you use CPUParticles instead of GPUParticles in web exports.

> I dont know if there is any technical limitations for Godot 4 to support webgl 1.0

Yes, GLES 2.0 would need to be reimplemented for WebGL 1.0 to be supported again.

**Comment by adamscott** [comment]
Currently, issues go away on macOS and iOS when you're using single threaded exports.
